### PR TITLE
Feat/guppy explorer

### PR DESCRIPTION
**Pull request by qingyashu**
Use guppy for data explorer instead of arranger. Portal config like this (under `dataExplorerConfig` block): 
```
"guppyConfig": {
      "dataType": "subject",
      "fieldMapping": [
        { "field": "project", "name": "Project" },
        { "field": "study", "name": "Study" },
        { "field": "race", "name": "Race" },
        { "field": "ethnicity", "name": "Ethnicity" },
        { "field": "gender", "name": "Gender" },
        { "field": "vital_status", "name": "Vital Status" },
        { "field": "whatever_lab_result_value", "name": "Lab Result Value" },
        { "field": "file_count", "name": "File Count" },
        { "field": "file_type", "name": "File Type" },
        { "field": "file_format", "name": "File Format" }
      ],
      "manifestMapping": {
        "resourceIndexType": "file",
        "resourceIdField": "file_id",
        "referenceIdFieldInResourceIndex": "subject_id",
        "referenceIdFieldInDataIndex": "subject_id"
      }
    },
    "charts": {
      "project": {
        "chartType": "count",
        "title": "Projects"
      },
      "study": {
        "chartType": "count",
        "title": "Studies"
      },
      "file_type": {
        "chartType": "count",
        "title": "File Types"
      },
      "ethnicity": {
        "chartType": "stackedBar",
        "title": "Ethnicity"
      },
      "gender": {
        "chartType": "pie",
        "title": "Gender"
      },
      "race": {
        "chartType": "pie",
        "title": "Race"
      },
      "vital_status": {
        "chartType": "bar",
        "title": "Vital Status"
      }
    },
    "filters": {
      "tabs": [
        {
          "title": "Project",
          "fields": [
            "project",
            "study"
          ]
        },
        {
          "title": "Subject",
          "fields": [
            "race",
            "ethnicity",
            "gender",
            "vital_status"
          ]
        },
        {
          "title": "File",
          "fields": [
            "file_type"
          ]
        }
      ]
    },
    "buttons": [
      {
        "enabled": true,
        "type": "data",
        "title": "Download All Data",
        "leftIcon": "user",
        "rightIcon": "download",
        "fileName": "data.json"
      },
      {
        "enabled": true,
        "type": "manifest",
        "title": "Download Manifest",
        "leftIcon": "datafile",
        "rightIcon": "download",
        "fileName": "manifest.json"
      }
    ],
    "table": {
      "enabled": true,
      "fields": [
        "project",
        "study",
        "race",
        "ethnicity",
        "gender",
        "vital_status",
        "whatever_lab_result_value",
        "file_count",
        "file_type",
        "file_format"
      ]
    }
```
Notice that `filters` and `charts` are same as previous cofnig, and for `table` there's extra `table.fields` block. For backward compatibility, if there's no "guppyConfig" block in portal config, data explorer will still use arranger for data explorer page by reading `arrangerConfig` and related config blocks. 

### New Features
  - Use guppy for data explorer instead of arranger

### CODE REVIEW

File: data/config/default.json [hunk 1]
```diff
@@ -254,15 +254,48 @@
...
+        "whatever_lab_result_value",
+        "file_count",
+        "file_type",
+        "file_format"
+      ]
+    },
+    "guppyConfig": {
+      "dataType": "subject",
+      "fieldMapping": [
+        { "field": "project", "name": "Project" },
```
Comment (thanh-nguyen-dang) [inline data/config/default.json]: Should we automate this part to only configure the special mapping?
I mean if it is only `project -> Project` or `vital_status -> Vital Status` you don't need to put it in the fieldMapping.

Comment (qingyashu) [reply data/config/default.json]: I think sometimes some field name mapping will need more caring, like `thrpyv`, but yeah automation is a good idea, I think we could combine both automation and customized names

File: data/config/ndh.json [hunk 2]
```diff
@@ -123,7 +123,7 @@
     },
     "login": {
       "title": "NIAID Data Hub",
-      "subTitle": "Explore, Analyze, and Share Data",
+      "subTitle": "search, compare, and download data",
```
Comment (abgeorge7) [inline data/config/ndh.json]: can you change this back to "Explore, Analyze, and Share Data" ?

File: data/config/ndh.json [hunk 3]
```diff
@@ -194,17 +194,31 @@
...
-            "therapy",
-            "therapyv",
             "hiv_status"
           ]
         }
       ]
     },
     "table": {
-      "enabled": false
+      "enabled": true,
```
Comment (abgeorge7) [inline data/config/ndh.json]: and this to false?

File: src/GuppyDataExplorer/ExplorerTable/index.jsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    const { totalCount } = this.props;
+    const { pageSize } = this.state;
+    const totalPages = Math.floor(totalCount / pageSize) + ((totalCount % pageSize === 0) ? 0 : 1);
+    const SCROLL_SIZE = 10000;
+    const visiblePages = Math.min(totalPages, Math.round((SCROLL_SIZE / pageSize) + 0.49));
+    const start = (this.state.currentPage * this.state.pageSize) + 1;
+    const end = (this.state.currentPage + 1) * this.state.pageSize;
+    return (
+      <div className={`explorer-table ${this.props.className}`}>
+        <p className='explorer-table__description'>{`Showing ${start} - ${end} of ${totalCount} cases`}</p>
```
Comment (abgeorge7) [inline src/GuppyDataExplorer/ExplorerTable/index.jsx]: "cases" here shouldn't be hardcoded - for example, NIAID uses "subjects"

File: data/config/ndh.json (hunk 3 above)
Comment (qingyashu) [reply data/config/ndh.json]: Sure~

File: data/config/ndh.json (hunk 2 above)
Comment (qingyashu) [reply data/config/ndh.json]: Prod use this so I copied them

Comment (abgeorge7) [reply data/config/ndh.json]: :O I'll change prod - they should all be "Explore, Analyze, and Share Data" now

**Pull request merged**